src/stactools/sentinel1/grd/properties.py

```diff
@@ -61,21 +76,37 @@ def fill_sat_properties(sat_ext, href):
         root.findall(".//safe:relativeOrbitNumber")[0].text)
 
 
-def fill_proj_properties(proj_ext, meta_links, product_meta):
+def fill_proj_properties(
+    proj_ext,
+    meta_links,
+    product_meta,
+    read_href_modifier: Optional[ReadHrefModifier] = None,
+):
     """Fills the properties for SAR.
 
     Based on the sar Extension.py
 
     Args:
-        input_ext (pystac.extensions.sar.SarExtension): The extension to be populated.
-        href (str): The HREF to the scene, this is expected to be an XML file.
+        proj_ext (pystac.extensions.sar.SarExtension): The extension to be populated.
+        read_href_modifier: A function that takes an HREF and returns a modified HREF.
+            This can be used to modify a HREF to make it readable, e.g. appending
+            an Azure SAS token or creating a signed URL.
 
     Returns:
         pystac.Asset: An asset with the SAR relevant properties.
     """
     # Read meta file
     links = meta_links.create_product_asset()
-    root = XmlElement.from_file(links[0][1].href)
+    root = XmlElement.from_file(links[0][1].href, read_href_modifier)
+
+    # Compute transform
+    head, _ = os.path.split(meta_links.product_metadata_href)
+    src = rasterio.open(f'{head}/{meta_links.grouped_hrefs["measurement"][0]}')
+    gcps = src.get_gcps()
+    s1_transform = rasterio.transform.from_gcps(gcps[0])
```

Comment:
One of these calls GDAL in the background and prints a GDAL ERROR to the console. I cannot suppress this error (that doesn't stop the code working).